### PR TITLE
add: #729 Keptn CLI Autocompletion Instructions

### LINKED ISSUE #729: Provide docs for auto-completion feature in CLI
Similar to https://kubernetes.io/docs/tasks/tools/install-kubectl/#enabling-shell-autocompletion we need a documentation on how to use the auto-completion feature in the CLI.

Currently, it is not clear how to use it, as e.g., this issue shows: https://github.com/keptn/keptn/issues/3212

### PR DESCRIPTION
**Pull request by christian-kreuzberger-dtx**
Signed-off-by: Hemanth Krishna (DarthBenro008) <hkpdev008@gmail.com>
Signed-off-by: DarthBenro008 <hkpdev008@gmail.com>

Fixes #729 
superseedes PR https://github.com/keptn/keptn.github.io/pull/758

I have squashed the commits and removed the generated files

### CODE REVIEW

File: content/docs/0.8.x/reference/cli/_index.md [hunk 1]
```diff
@@ -198,3 +198,144 @@ The `KEPTNCONFIG` environment variable holds a config file that contains the cre
 * If `KEPTNCONFIG` environment variable is defined, the Keptn CLI reads the credentials from it. 
 
 * If the credentials are not found in the config file, it will fall back to the credential manager.
+
+## Enabling shell autocompetion of Keptn CLI
```
Comment (johannes-b) [inline content/docs/0.8.x/reference/cli/_index.md:202]: autocompletion

File: content/docs/0.8.x/reference/cli/_index.md [hunk 2]
```diff
@@ -198,3 +198,144 @@ The `KEPTNCONFIG` environment variable holds a config file that contains the cre
...
+
+To find out, reload your shell and run `type _init_completion`. If the command succeeds, you're already set, otherwise add the following to your ~/.bashrc file:
+
+```bash
+source /usr/share/bash-completion/bash_completion
+```
+
+Reload your shell and verify that bash-completion is correctly installed by typing `type _init_completion`.
+
+### Enable Keptn CLI autocompletion for linux (bash)
```
Comment (johannes-b) [inline content/docs/0.8.x/reference/cli/_index.md:228]: **L**inux

File: content/docs/0.8.x/reference/cli/_index.md [hunk 3]
```diff
@@ -198,3 +198,144 @@ The `KEPTNCONFIG` environment variable holds a config file that contains the cre
...
+```
+
+If you have an alias for Keptn CLI, you can extend shell completion to work with that alias:
+
+```bash
+echo 'alias kc=keptn' >>~/.bashrc
+echo 'complete -F __start_keptn kc' >>~/.bashrc
+```
+
+After reloading your shell, Keptn CLI autocompletion will be enabled succesfully.
```
Comment (johannes-b) [inline content/docs/0.8.x/reference/cli/_index.md:251]: succe**s**sfully

File: content/docs/0.8.x/reference/cli/_index.md [hunk 4]
```diff
@@ -198,3 +198,144 @@ The `KEPTNCONFIG` environment variable holds a config file that contains the cre
...
+```
+
+- If you have an alias for Keptn CLI, you can extend shell completion to work with that alias:
+
+```bash
+echo 'alias kc=keptn' >>~/.bash_profile
+echo 'complete -F keptn kc' >>~/.bash_profile
+```
+
+After reloading your shell, Keptn CLI autocompletion will be enabled succesfully.
```
Comment (johannes-b) [inline content/docs/0.8.x/reference/cli/_index.md:304]: succe**s**sfully

File: content/docs/0.8.x/reference/cli/_index.md [hunk 5]
```diff
@@ -198,3 +198,144 @@ The `KEPTNCONFIG` environment variable holds a config file that contains the cre
...
+```
+
+If you have an alias for Keptn CLI, you can extend shell completion to work with it as follows:
+
+```bash
+echo 'alias kc=keptn' >>~/.zshrc
+echo 'complete -F __start_keptn kc' >>~/.zshrc
+```
+
+After reloading your shell, Keptn CLI autocompletion will be enabled succesfully.
```
Comment (johannes-b) [inline content/docs/0.8.x/reference/cli/_index.md:332]: succe**s**sfully

File: content/docs/0.8.x/reference/cli/_index.md [hunk 6]
```diff
@@ -198,3 +198,144 @@ The `KEPTNCONFIG` environment variable holds a config file that contains the cre
...
+
+The Keptn CLI completion script for Bash can be generated with `keptn completion bash`. Sourcing this script in your shell enables Keptn CLI autocompletion.
+
+However, the Keptn CLI autocompletion script depends on bash-completion which you thus have to previously install.
+
+> Warning: There are two versions of bash-completion, v1 and v2. V1 is for Bash 3.2 (which is the default on macOS), and v2 is for Bash 4.1+. The Keptn CLI autocompletion script doesn't work correctly with bash-completion v1 and Bash 3.2. It requires bash-completion v2 and Bash 4.1+. Thus, to be able to correctly use the Keptn CLI autocompletion on macOS, you have to install and use Bash 4.1+ (instructions). The following instructions assume that you use Bash 4.1+ (that is, any Bash version of 4.1 or newer).
+
+### Install bash-completion for MacOS
+
+Note: As mentioned, these instructions assume you use Bash 4.1+, which means you will install bash-completion v2 (in contrast to Bash 3.2 and bash-completion v1, in which case Keptn CLI completion won't work).
```
Comment (johannes-b) [inline content/docs/0.8.x/reference/cli/_index.md:264]: **Note:** (please in bold)

**Review by johannes-b — APPROVED** [review]
lgtm

**Comment by christian-kreuzberger-dtx** [comment]
Thanks @DarthBenro008

**Pull request merged**